tests/phpunit/tests/interactivity-api/wpInteractivityAPI.php

```diff
@@ -508,50 +508,130 @@ public function test_process_directives_doesnt_change_html_if_contains_unbalance
 	}
 
 	/**
-	 * Tests that the `process_directives` returns the same HTML if it finds an
-	 * SVG tag.
+	 * Tests that the `process_directives` process the HTML outside a SVG tag.
 	 *
-	 * @ticket 60356
+	 * @ticket 60517
 	 *
 	 * @covers ::process_directives
 	 */
-	public function test_process_directives_doesnt_change_html_if_contains_svgs() {
-		$this->interactivity->state( 'myPlugin', array( 'id' => 'some-id' ) );
+	public function test_process_directives_changes_html_if_contains_svgs() {
+		$this->interactivity->state(
+			'myPlugin',
+			array(
+				'id'    => 'some-id',
+				'width' => '100',
+			)
+		);
 		$html           = '
-			<div data-wp-bind--id="myPlugin::state.id">
-				<svg height="100" width="100">
+			<header>
+				<svg height="100" data-wp-bind--width="myPlugin::state.width">
 					<circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />
-				</svg> 
-			</div>
+				</svg>
+				<div data-wp-bind--id="myPlugin::state.id"></div>
+				<div data-wp-bind--id="myPlugin::state.width"></div>
+			</header>
 		';
 		$processed_html = $this->interactivity->process_directives( $html );
 		$p              = new WP_HTML_Tag_Processor( $processed_html );
-		$p->next_tag();
+		$p->next_tag( 'svg' );
+		$this->assertNull( $p->get_attribute( 'width' ) );
+		$p->next_tag( 'div' );
+		$this->assertEquals( 'some-id', $p->get_attribute( 'id' ) );
+		$p->next_tag( 'div' );
+		$this->assertEquals( '100', $p->get_attribute( 'id' ) );
+	}
+
+	/**
+	 * Tests that the `process_directives` does not process the HTML
+	 * inside SVG tags.
+	 *
+	 * @ticket 60517
+	 *
+	 * @covers ::process_directives
+	 */
+	public function test_process_directives_does_not_change_inner_html_in_svgs() {
+		$this->interactivity->state(
+			'myPlugin',
+			array(
+				'id' => 'some-id',
+			)
+		);
+		$html           = '
+			<header>
+				<svg height="100">
+					<circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />
+					<div data-wp-bind--id="myPlugin::state.id"></div>
```

Comment:
Why not an SVG element?

```suggestion
					<g data-wp-bind--id="myPlugin::state.id" />
```

Reply:
this is a good example of a bad example anyway 🙃 as the `div` here would close the open `SVG` element 🤦‍♂️

the full set of HTML tags which break out to the nearest integration point are:

> "b", "big", "blockquote", "body", "br", "center", "code", "dd", "div", "dl", "dt", "em", "embed", "h1", "h2", "h3", "h4", "h5", "h6", "head", "hr", "i", "img", "li", "listing", "menu", "meta", "nobr", "ol", "p", "pre", "ruby", "s", "small", "span", "strong", "strike", "sub", "sup", "table", "tt", "u", "ul", "var"

and some `FONT` start tags containing specific attributes. also some end tags.

https://html.spec.whatwg.org/#parsing-main-inforeign

the "integration point" thing is a bit more complicated than it appears at a glance. there can be a stack of open elements that zigzags between HTML and foreign content, and some SVG/MathML tags themselves create new integration points, so a `DIV` inside one of those regions won't close out the `SVG` - it will only close out everything up until that element.

Reply:
The idea was to use an element that we know the server directive processing would process. As far as I know, it should skip SVG tags and its inner accepted elements by default.

Reply:
@c4rl0sbr4v0 my point was mostly that in an HTML document, the `DIV` in this example closes out the `SVG`, and so the `skip_foreign_content()` function would fail here.

if you wanted to use an element that's neutral on this, it would be fine to use something like `<aside>` or `<data>` or `<abbr>`, as these and any other elements not listed above do not affect the foreign content stack.

Reply:
If they are edge cases, maybe is better to just use what should be inside a SVG tag.

> if you wanted to use an element that's neutral on this, it would be fine to use something like <aside> or <data> or <abbr>, as these and any other elements not listed above do not affect the foreign content stack.

You are becoming a huge expert in HTML processing @dmsnell 😃